src/auto/calendar.rs

```diff
@@ -200,9 +201,28 @@ impl<O: IsA<Calendar>> CalendarExt for O {
         }
     }
 
-    //fn set_detail_func<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, func: /*Unknown conversion*//*Unimplemented*/CalendarDetailFunc, data: P, destroy: /*Unknown conversion*//*Unimplemented*/DestroyNotify) {
-    //    unsafe { TODO: call ffi::gtk_calendar_set_detail_func() }
-    //}
+    fn set_detail_func<P: Fn(&Calendar, u32, u32, u32) -> String + 'static>(&self, func: P) {
```

Comment:
Why does this not return a `GString`, but for parameters we use `GString`? :)

Reply:
Because there for trampolines we decode that returning String instead GString better, don't remember reasoning.
So hack added in https://github.com/gtk-rs/gir/blob/master/src/codegen/return_value.rs#L19.

Reply:
.. and as you say in https://github.com/gtk-rs/gio/pull/183#discussion_r249974199 we need use `&str` as parameter instead GString.